pkg/argocd/secret.go

```diff
@@ -98,6 +102,49 @@ func CreateArgoSecret(ctx context.Context, config *rest.Config, namespace, passw
 	return err
 }
 
+func createRepoSecret(ctx context.Context, cluster *api.Cluster, clientset *kubernetes.Clientset, namespace string) error {
+	if cluster.GitRepo == nil || cluster.GitRepo.Url == nil {
+		return fmt.Errorf("no git repo information received from API for cluster '%s'", cluster.Id)
+	}
+	gitURL, err := url.Parse(*cluster.GitRepo.Url)
+	if err != nil {
+		return err
+	}
+
+	repoSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: argoRepoSecretName,
+			Labels: map[string]string{
+				"argocd.argoproj.io/secret-type": "repository",
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"type": []byte("git"),
+			"url":  []byte(gitURL.String()),
+		},
+	}
+
+	_, err = clientset.CoreV1().Secrets(namespace).Create(ctx, repoSecret, metav1.CreateOptions{})
+	klog.Infof("Created new Repo secret")
+
+	// Patch credential secret to also map to the URL
+
+	sshSecret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, argoSSHSecretName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	sshSecret.Data["url"] = []byte(gitURL.String())
+	_, err = clientset.CoreV1().Secrets(namespace).Update(ctx, sshSecret, metav1.UpdateOptions{})
```

Comment:
why do we do this here instead of doing it right when we create the SSH key?

Reply:
I didn't have the URL at the point in the control flow where the SSH key is created. I can look into restructuring things, but as it stands the cluster object is created after the SSH key is initially generated.